diff --git a/zip.h b/zip.h
index ba03160..f5fdb72 100644
--- a/zip.h
+++ b/zip.h
@@ -136,6 +136,10 @@ typedef struct iztimes {
    time_t ctime;                /* new creation time (!= Unix st.ctime) */
 } iztimes;
 
+struct zos_file_tag {                /* File Tag Attributes         */
+       unsigned short ft_ccsid;       /*   Character Set Id          */
+     };
+
 /* Lengths of headers after signatures in bytes */
 #define LOCHEAD 26
 #define CENHEAD 42
